src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -82,15 +82,16 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
 
             _formatters = formatters;
             _readerFactory = readerFactory.CreateReader;
-
-            if (loggerFactory != null)
-            {
-                _logger = loggerFactory.CreateLogger<BodyModelBinder>();
-            }
+            Logger = loggerFactory.CreateLogger(GetType());
 
             _options = options;
         }
 
+        /// <summary>
+        /// The <see cref="ILogger"/> used for logging in this binder.
+        /// </summary>
+        protected ILogger Logger { get; }
```

Comment:
Add `protected` properties only where we already have subclasses and the subclasses need the properties. Don't build infrastructure in case of _future_ requirements.